### ISSUE TITLE
Issue on removeAllListeners

### ISSUE THREAD

**Issue by jeromevalentin** [RangerMauve/mqtt-emitter#10]
Thanks for your library it helps a lot!

My application is using your library to manage its MQTT subscriptions since a long time.
Migrating to the last 4.0.0 leads me to get the following error when calling removeAllListeners:
```
TypeError: Cannot read property 'pattern' of undefined
```
In reality, I'm trying to clean up the subscription asap along the life of my application, but for safety I'm also cleaning up all subscriptions before closing the connection by calling removeAllListeners.

After investigations, it looks like you are not filtering the listeners that may be empty from the removeAllListeners implementation, leading to the error I encountered.

With the following implementation it works properly (I just filtered out the empty listeners from the topicStrings list construction).

```javascript
/**
 * Removes all listeners for this type of topic.
 * @param  {String}      topic Topic pattern to unsubscribe from
 * @return {MQTTEmitter}       Returns self for use in chaining
 */
function removeAllListeners (topic) {
  if (topic) {
    var topicString = MQTTPattern.clean(topic)
    var listeners = this._listeners.get(topicString)

    if (!listeners.length) return this

    this._listeners.set(topicString, [])
    this.onremove(topicString)
  } else {
    var topicStrings = this._listeners.query('#').filter(function (listeners) {
      return listeners && listeners.length > 0
    }).map(function (listeners) {
      return MQTTPattern.clean(listeners[0].pattern)
    })

    for (var i in topicStrings) {
      this.onremove(topicStrings[i])
    }

    this._listeners = new MQTTStore()
  }

  return this
}
```

Hope it helps,
Regards,
Jérôme

**Comment by RangerMauve** [comment]
Would you be up for submitting a pull request for this fix? :D

**Comment by jeromevalentin** [comment]
Sure! Tomorrow ;)

**Comment by RangerMauve** [comment]
Fixed in 4.0.1
